core/context.go

```diff
@@ -199,6 +199,9 @@ func CloseContextRootFinalizer(ctx *sql.Context) error {
 	}
 	if newRoot != nil {
 		if err = session.SetWorkingRoot(ctx, ctx.GetCurrentDatabase(), newRoot); err != nil {
```

Comment:
This is fine for the time being, but please add a TODO here. We need a way to see if the session has a writeable working root (new interface method on session probably), and avoid setting it if so.